cmd/go-cacher/cacher.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2023 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// The go-cacher binary is a cacher helper program that cmd/go can use.
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+
+	"github.com/bradfitz/go-tool-cache/cacheproc"
+	"github.com/bradfitz/go-tool-cache/cachers"
+)
+
+const defaultCacheKey = "v1"
+
+var (
+	serverBase = flag.String("cache-server", "", "optional cache server HTTP prefix (scheme and authority only); should be low latency. empty means to not use one.")
+	verbose    = flag.Bool("verbose", false, "be verbose")
+)
+
+func getAwsConfigFromEnv() (*aws.Config, error) {
+	// read from env
+	awsRegion, awsRegionOk := os.LookupEnv("GOCACHE_AWS_REGION")
+	if !awsRegionOk {
+		return nil, nil
+	}
+	accessKey, accessKeyOk := os.LookupEnv("GOCACHE_AWS_ACCESS_KEY")
+	secretAccessKey, secretKeyOk := os.LookupEnv("GOCACHE_AWS_SECRET_ACCESS_KEY")
+	if accessKeyOk && secretKeyOk {
+		cfg, err := config.LoadDefaultConfig(context.TODO(),
+			config.WithRegion(awsRegion),
+			config.WithCredentialsProvider(credentials.StaticCredentialsProvider{
+				Value: aws.Credentials{
+					AccessKeyID:     accessKey,
+					SecretAccessKey: secretAccessKey,
+				},
+			}))
+		if err != nil {
+			return nil, err
+		}
+		return &cfg, nil
+	}
+	credsProfile, credsProfileOk := os.LookupEnv("GOCACHE_CREDS_PROFILE")
+	if credsProfileOk {
+		cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(awsRegion), config.WithSharedConfigProfile(credsProfile))
+		if err != nil {
+			return nil, err
+		}
+		return &cfg, nil
+	}
+	return nil, nil
+}
+
+func maybeS3Cache() (cachers.RemoteCache, error) {
+	awsConfig, err := getAwsConfigFromEnv()
+	if err != nil {
+		return nil, err
+	}
+	bucket, ok := os.LookupEnv("GOCACHE_S3_BUCKET")
+	if !ok || awsConfig == nil {
+		// We need at least name of bucket and valid aws config
+		return nil, nil
+	}
+	cacheKey := os.Getenv("GOCACHE_CACHE_KEY")
```

Comment:
There are like 6 env vars of this sort scattered all over, I would group them together in a const and document them if needed so it would be easily visible

Reply:
<img width="984" alt="image" src="https://github.com/or-shachar/go-tool-cache/assets/2962963/eb294583-8080-4b93-8aeb-b9a536e0d093">
Something like this?